azure/services/virtualnetworks/virtualnetworks.go

```diff
@@ -66,6 +66,9 @@ func (s *Service) Reconcile(ctx context.Context) error {
 	defer cancel()
 
 	vnetSpec := s.Scope.VNetSpec()
+	if vnetSpec == nil {
+		return nil
+	}
 
 	result, err := s.CreateResource(ctx, vnetSpec, serviceName)
 	s.Scope.UpdatePutStatus(infrav1.VNetReadyCondition, serviceName, err)
```

Comment:
looks like we'd still be updating the status here when the vnet is unmanaged

Reply:
One suggestion: 

 to avoid the extra GET by calling IsManaged below, we should be able to use the information we have below in Tags to determine if the vnet is managed:

on line 83:
```
managed := tags.HasOwned(s.Scope.ClusterName()), nil
```

then below on line 90:

```
if managed {
    s.Scope.UpdatePutStatus(infrav1.VNetReadyCondition, serviceName, err)
}
return err
```

Reply:
Sure I'll remove that status update in that case. And wdym by an extra GET? IsVnetManaged() just does the same thing right? 
``` // IsVnetManaged returns true if the vnet is managed.
func (s *ClusterScope) IsVnetManaged() bool {
	return s.Vnet().ID == "" || s.Vnet().Tags.HasOwned(s.ClusterName())
}
```

Reply:
`IsVnetManaged` would work too as long as it's run after line 82 so we've already set the tags